src/main/java/com/iabtcf/v2/BitVectorTCModelV2.java

```diff
@@ -0,0 +1,348 @@
+    private final Set<Integer> customPurposesConsent;
+    private final Set<Integer> customPurposesLITransparency;
+
+    private BitVectorTCModelV2(BitVector coreBitVector) {
+        this.version = coreBitVector.readUnsignedInt(VERSION_OFFSET, TINY_INT.length());
+        this.consentRecordCreated =
+                coreBitVector.readInstantFromDeciSecond(CREATED_OFFSET, EPOCH_TIME.length());
+        this.consentRecordLastUpdated =
+                coreBitVector.readInstantFromDeciSecond(LAST_UPDATED_OFFSET, EPOCH_TIME.length());
+        this.consentManagerProviderId = coreBitVector.readUnsignedInt(CMP_ID_OFFSET, SHORT.length());
+        this.consentManagerProviderVersion =
+                coreBitVector.readUnsignedInt(CMP_VERSION_OFFSET, SHORT.length());
+        this.consentScreen = coreBitVector.readUnsignedInt(CONSENT_SCREEN_OFFSET, TINY_INT.length());
+        this.consentLanguage = coreBitVector.readStr(CONSENT_LANGUAGE_OFFSET, CHAR.length() * 2);
+        this.vendorListVersion =
+                coreBitVector.readUnsignedInt(VENDOR_LIST_VERSION_OFFSET, SHORT.length());
+        this.policyVersion = coreBitVector.readUnsignedInt(TCF_POLICY_VERSION_OFFSET, TINY_INT.length());
+        this.isServiceSpecific = coreBitVector.readBit(IS_SERVICE_SPECIFIC_OFFSET);
+        this.useNonStandardStacks = coreBitVector.readBit(USE_NON_STANDARD_STACKS_OFFSET);
+        this.specialFeaturesOptInts =
+                fillSet(SPECIAL_FEATURE_OPT_INS_OFFSET, SPECIAL_FEATURE_OPT_INS_LENGTH, coreBitVector);
+        this.purposesConsent = fillSet(PURPOSES_CONSENT_OFFSET, PURPOSES_CONSENT_LENGTH, coreBitVector);
+        this.purposesLITransparency =
+                fillSet(PURPOSE_LI_TRANSPARENCY_OFFSET, PURPOSE_LI_TRANSPARENCY_LENGTH, coreBitVector);
+        this.isPurposeOneTreatment = coreBitVector.readBit(PURPOSE_ONE_TREATMENT_OFFSET);
+        this.publisherCountryCode = coreBitVector.readStr(PUBLISHER_CC_OFFSET, CHAR.length() * 2);
+
+        int currentPointer = PUBLISHER_CC_OFFSET + (CHAR.length() * 2); // publisher cc offset
+        this.vendorConsents = new TreeSet<>();
+        currentPointer = this.fetchSet(this.vendorConsents, currentPointer, coreBitVector);
+
+        this.vendorLegitimateInterests = new TreeSet<>();
+        currentPointer = this.fetchSet(this.vendorLegitimateInterests, currentPointer, coreBitVector);
+
+        this.publisherRestrictions = new ArrayList<>();
+        this.fillPublisherRestrictions(publisherRestrictions, currentPointer, coreBitVector);
+
+        this.disclosedVendors = new TreeSet<>();
+        this.allowedVendors = new TreeSet<>();
+        this.publisherPurposesConsent = new TreeSet<>();
+        this.publisherPurposesLITransparency = new TreeSet<>();
+        this.customPurposesConsent = new TreeSet<>();
+        this.customPurposesLITransparency = new TreeSet<>();
+    }
+
+    private BitVectorTCModelV2(BitVector coreVector, BitVector secondPart) {
```

Comment:
If I understand the spec correct there are four segements possible, where the core segement the implementation is correct.

I was thinking here that maybe a dynamic parameters are an better solution.

    private BitVectorTCModelV2(BitVector coreVector, BitVector... remainingBitVectors) {
        this(coreVector);

        for (BitVector bitVector : remainingBitVectors) {
            fillRemainingVector(bitVector);
        }
    }

Maybe the segments should check for plausibilty.